pkg/volume/csi/csi_plugin.go

```diff
@@ -201,20 +206,37 @@ func (p *csiPlugin) GetPluginName() string {
 // GetvolumeName returns a concatenated string of CSIVolumeSource.Driver<volNameSe>CSIVolumeSource.VolumeHandle
 // That string value is used in Detach() to extract driver name and volumeName.
 func (p *csiPlugin) GetVolumeName(spec *volume.Spec) (string, error) {
-	csi, err := getCSISourceFromSpec(spec)
+	var driver, handle string
+	volSource, pvSource, err := getSourceFromSpec(spec)
 	if err != nil {
 		klog.Error(log("plugin.GetVolumeName failed to extract volume source from spec: %v", err))
 		return "", err
 	}
+	if volSource != nil && p.inlineFeatureEnabled {
+		driver = volSource.Driver
+		if volSource.VolumeHandle != nil {
+			handle = *volSource.VolumeHandle
+		}
+	} else if pvSource != nil {
+		driver = pvSource.Driver
+		handle = pvSource.VolumeHandle
+	} else {
+		return "", errors.New("spec missing CSI volume source")
+	}
 
 	// return driverName<separator>volumeHandle
-	return fmt.Sprintf("%s%s%s", csi.Driver, volNameSep, csi.VolumeHandle), nil
+	return fmt.Sprintf("%s%s%s", driver, volNameSep, handle), nil
 }
 
 func (p *csiPlugin) CanSupport(spec *volume.Spec) bool {
 	// TODO (vladimirvivien) CanSupport should also take into account
 	// the availability/registration of specified Driver in the volume source
-	return spec.PersistentVolume != nil && spec.PersistentVolume.Spec.CSI != nil
+	if p.inlineFeatureEnabled {
+		return (spec.PersistentVolume != nil && spec.PersistentVolume.Spec.CSI != nil) ||
```

Comment:
if we're doing inline why are we checking whether PersistentVolume is not nil either? We specifically want Persistent volume to be nil actually?